src/Symfony/Component/CssSelector/Node/MatchingNode.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\CssSelector\Node;
+
+/**
+ * Represents a "<selector>:is(<subSelectorList>)" node.
+ *
+ * This component is a port of the Python cssselect library,
+ * which is copyright Ian Bicking, @see https://github.com/SimonSapin/cssselect.
+ *
+ * @author Hubert Lenoir <lenoir.hubert@gmail.com>
+ *
+ * @internal
+ */
+class MatchingNode extends AbstractNode
+{
+    /**
+     * @param array<NodeInterface> $arguments
+     */
+    public function __construct(
+        public readonly NodeInterface $selector,
+        public readonly array $arguments = []
+    ) {
+    }
+
+    public function getSpecificity(): Specificity
```

Comment:
Does this class represent `<selector>:is(<subSelectorList>)` or only `:is(<subSelectorList>)` ? In the first case, the specificity of `<selector>` needs to be taken into account

Reply:
This class includes the selector. I made changes to add selector specificity.